htmd/smallmol/chemlab/builder.py

```diff
@@ -0,0 +1,406 @@
+from htmd.smallmol.smallmol import SmallMol
+from htmd.smallmol.chemlab.periodictable import PeriodicTable, _heavy_atoms, _hetero_atoms, _hybridizations_IdxToType
+import numpy as np
+from htmd.smallmol.util import _getPerpendicular, _normalizeVector, _getRotationMatrix
+
+
+class Builder:
+    """
+    A class to modify a SmallMol object: adding and removing hydrogens; creation of ligands from scracth
+
+    Parameters
+    ----------
+    smallmol: htmd.smallmol.smallmol.SmallMol
+        The SmallMol object
+        Default: None
+    checkInitialConformer: bool
+        If True, a conformation is generated if None are found. (designed espcially for smiles)
+        Default: True
+
+    Example
+    -------
+    >>> sm = SmallMol('CCC')
+    >>> sm.coords[0], sm.numAtoms
+   ( array([[ 0.],[ 0.],[ 0.]]), 11)
+   >>> B = Builder(sm)
+   >>> sm_new = B.getSmallMol()
+   >>> sm_new.coords[0]
+   array([[ 1.1701015 ],[-0.21571056],[-0.21615667]])
+   >>> B.removeHydrogens()
+   >>> B.getSmallMol().numAtoms
+   3
+    """
+
+    def __init__(self, smallmol=None, checkInitialConformer=True):
+
+        self.smallmol = smallmol
+        self.periodictable = PeriodicTable()
+
+        if checkInitialConformer and smallmol is not None:
+            self._prepareConformer()
+
+
+    def loadMol(self, smallmol, checkInitialConformer=True):
+        """
+        Loads the SmallMol object. The additional argument checkInitialConformer ensure that one valid conformer exists
+
+        Parameters
+        ---------
+        smallmol: htmd.smallmol.smallmol.SmallMol
+            The SmallMol object
+        checkInitialConformer: bool
+            If True, a conformer is generated if only not valid are found in the smallmol object
+            Default: True
+        """
+
+        if not isinstance(smallmol, SmallMol):
+            raise ValueError('The argument tyep {} is not accepted. Only SmallMol object. '.format(smallmol))
+
+        self.smallmol = smallmol.copy()
+
+        if checkInitialConformer:
+            self._prepareConformer()
+
+    def _prepareConformer(self):
+        """
+        Generate a conformer and stores the coords in the SmallMol object
+        """
+
+        from rdkit.Chem import AllChem
+
+        sm = self.smallmol
+        mol = sm.toRdkitMol(includeConformer=True)
+
+        AllChem.EmbedMolecule(mol, AllChem.ETKDG())
+
+        conf = mol.GetConformer(0)
+        coords = conf.GetPositions()
+        coords = coords[:, :, np.newaxis]
+        sm.coords = coords
+
+
+    def addHydrogens(self, onlyExplicit=False):
+        """
+        Adds the hydrogens to fill up the valence of the atoms. With onlyExplicit is possible to add only the polar ones
+
+        Parameters
+        ----------
+        onlyExplicit: bool
+            If True, only the polar hydrogens will be added
+            Default: False
```

Comment:
Small detail but there is no need to document the default values as they can be seen in the function signature. So better avoid documenting them.

Reply:
removed all the documented default value in the whole smallmol module